core/src/smartcontracts/wasm/mod.rs

```diff
@@ -223,90 +323,54 @@ pub struct Runtime<'wrld> {
 }
 
 impl<'wrld> Runtime<'wrld> {
-    fn create_store(&self, state: State<'wrld>) -> Result<Store<State<'wrld>>> {
+    fn create_store(&self, state: State<'wrld>) -> Store<State<'wrld>> {
         let mut store = Store::new(&self.engine, state);
 
         store.limiter(|stat| &mut stat.store_limits);
         store
             .add_fuel(self.config.fuel_limit)
-            .map_err(|err| Error::Instantiation(eyre!(Box::new(err))))?;
+            .expect("Wasm Runtime config is malformed, this is a bug");
 
-        Ok(store)
+        store
     }
 
     fn create_smart_contract(
         &self,
         store: &mut Store<State<'wrld>>,
         bytes: impl AsRef<[u8]>,
     ) -> Result<wasmtime::Instance> {
-        load_module(&self.engine, bytes).and_then(|module| {
-            self.linker
-                .instantiate(store, &module)
-                .map_err(|err| Error::Instantiation(eyre!(Box::new(err))))
-        })
+        let module = load_module(&self.engine, bytes)?;
+        self.instantiate_module(&module, store).map_err(Into::into)
     }
 
-    /// Host defined function which executes query. When calling this function, module
-    /// serializes query to linear memory and provides offset and length as parameters
-    ///
-    /// # Warning
-    ///
-    /// This function doesn't take ownership of the provided allocation
-    /// but it does transfer ownership of the result to the caller
-    ///
-    /// # Errors
-    ///
-    /// If decoding or execution of the query fails
-    fn execute_query(
-        mut caller: Caller<State>,
-        offset: WasmUsize,
-        len: WasmUsize,
-    ) -> Result<WasmUsize, Trap> {
-        let alloc_fn = Self::get_alloc_fn(&mut caller)?;
-        let memory = Self::get_memory(&mut caller)?;
-
-        let query: QueryBox = Self::decode_from_memory(&memory, &caller, offset, len)?;
-
-        let wsv = caller.data().wsv;
+    /// Execute `query` on host
+    #[allow(clippy::needless_pass_by_value)]
+    #[codec::wrap]
+    fn execute_query(query: QueryBox, state: &State) -> Result<Value, QueryExecutionFailure> {
```

Comment:
```suggestion
    fn execute_query(state: &State, query: QueryBox) -> Result<Value, QueryExecutionFailure> {
```

I would rather reverse it and say that args[0] is always `State`

Reply:
irrelevant